synchronizer/l1_check_block/integration.go

```diff
@@ -80,9 +88,46 @@ func (v *L1BlockCheckerIntegration) OnStartL2Sync(ctx context.Context) bool {
 	return v.checkBackgroundResult(ctx, "before start 2 sync")
 }
 
-// OnCheckReorg is a method that is called when a reorg is checked
-func (v *L1BlockCheckerIntegration) OnCheckReorg(ctx context.Context, latestBlock *state.Block) bool {
-	return v.checkBackgroundResult(ctx, "OnCheckReorg")
+// OnResetState is a method that is called after a resetState
+func (v *L1BlockCheckerIntegration) OnResetState(ctx context.Context) {
+	log.Infof("%s L1BlockChecker: after a resetState relaunch background process", logPrefix)
+	v.launch(ctx)
+}
+
+// CheckReorgWrapper is a wrapper over reorg function of synchronizer.
+// it checks the result of the function and the result of background process and decides which return
+func (v *L1BlockCheckerIntegration) CheckReorgWrapper(ctx context.Context, reorgFirstBlockOk *state.Block, errReportedByReorgFunc error) (*state.Block, error) {
+	resultBackground := v.getMergedResults()
+	if resultBackground != nil && resultBackground.ReorgDetected {
+		// Background process detected a reorg, decide which return
+		firstOkBlockBackgroundCheck, err := v.state.GetPreviousBlockToBlockNumber(ctx, resultBackground.BlockNumber, nil)
+		if err != nil {
+			log.Warnf("%s Error getting previous block to block number where a reorg have been detected %d: %s. So we reorgFunc values", logPrefix, resultBackground.BlockNumber, err)
```

Comment:
This seems to be an error
```suggestion
			log.Errorf("%s Error getting previous block to block number where a reorg have been detected %d: %s. So we reorgFunc values", logPrefix, resultBackground.BlockNumber, err)
```

Reply:
It's not an error, is just because is  asynchronuous process and have missed some reset done in some point, it need to resync with stateDB